tests/test_vector.py

```diff
@@ -378,44 +378,526 @@ def test_extract(vs, ws, vms, sms):
+                        lambda m, x: x[index].new(mask=m, replace=False, dtype=float),
+                        (vm.S, w),
+                        (dvm.S, dw),
+                        errors=True,
+                    )
+                    compare(h5, (vm.S, v.dup(), w), (dvm.S, dv.dup(), dw))
+                    compare(
+                        h5,
+                        (vm.S, v.dup(dtype=float), w),
+                        (dvm.S, dv.dup(dtype=float), dw),
+                    )
+                    compare(
+                        lambda m, x: x[index].new(mask=m, accum=gb.binary.plus),
+                        (vm.S, w),
+                        (dvm.S, dw),
+                        errors=True,
+                    )
+                    compare(
+                        lambda m, x: x[index].new(
+                            mask=m, accum=gb.binary.plus, dtype=float
+                        ),
+                        (vm.S, w),
+                        (dvm.S, dw),
+                        errors=True,
+                    )
+                    compare(i1, (vm.S, v.dup(), w), (dvm.S, dv.dup(), dw))
+                    compare(
+                        i1,
+                        (vm.S, v.dup(dtype=float), w),
+                        (dvm.S, dv.dup(dtype=float), dw),
+                    )
+                    compare(
+                        lambda m, x: x[index].new(
+                            mask=m, accum=gb.binary.plus, replace=True
+                        ),
+                        (vm.S, w),
+                        (dvm.S, dw),
+                        errors=True,
+                    )
+                    compare(
+                        lambda m, x: x[index].new(
+                            mask=m, accum=gb.binary.plus, replace=True, dtype=float
+                        ),
+                        (vm.S, w),
+                        (dvm.S, dw),
+                        errors=True,
+                    )
+                    compare(i2, (vm.S, v.dup(), w), (dvm.S, dv.dup(), dw))
+                    compare(
+                        i2,
+                        (vm.S, v.dup(dtype=float), w),
+                        (dvm.S, dv.dup(dtype=float), dw),
+                    )
+                    compare(
+                        lambda m, x: x[index].new(
+                            mask=m, accum=gb.binary.plus, replace=False
+                        ),
+                        (vm.S, w),
+                        (dvm.S, dw),
+                        errors=True,
+                    )
+                    compare(
+                        lambda m, x: x[index].new(
+                            mask=m, accum=gb.binary.plus, replace=False, dtype=float
+                        ),
+                        (vm.S, w),
+                        (dvm.S, dw),
+                        errors=True,
+                    )
+
+
+def test_assign(vs, ws, vms, sms):
+
+    test_replace_true = False
```

Comment:
I've switched of the case of `replace=True` for now.
The problem here is that since the chunk-access patterns during updating are index-dependent, if none of the input indices refer to any of the elements in one or more chunks of the `Vector` being updated, then those ignored chunks' values are not cleared before the assignment. 

Not really sure how to fix this.  `dask` seems to be smart enough to know which chunks need updating and those that do not (which it ignores).